cassandra/protocol.py

```diff
@@ -122,32 +127,35 @@ def to_exception(self):
         return self
 
     class Codec(object):
-        opcode = 0x00
         error_decoders = None
 
         def __init__(self, error_decoders):
             self.error_decoders = {opcode: decoder for opcode, decoder in error_decoders}
 
-        def decode(self, f, protocol_version, *args, **kwargs):
+        def decode(self, f, protocol_version, *args):
             code = read_int(f)
             msg = read_string(f)
             error_decoder = self.error_decoders.get(code)
             if error_decoder:
-                return error_decoder(f, protocol_version, msg, **kwargs)
+                return error_decoder.decode(f, protocol_version, msg)
             return ErrorMessage(code=code, message=msg, info=None)
 
 
 class ErrorMessageSub(ErrorMessage):
     error_code = None
 
-    @staticmethod
-    def decode_error_info(f, protocol_version, **kwargs):
-        pass
+    class Codec(object):
+        error_class = None
+
+        def __init__(self, error_class):
+            self.error_class = error_class
```

Comment:
This is because I can't reference the outer class :(. We could also patch the class *after* it's definition with something like `MyErrorClass.Codec.error_class = MyErrorClass`

Reply:
I like this approach Jim. +1 to apply it.